### PR TITLE
claim determines owner DID

### PR DESCRIPTION
**Pull request by arty-name**
https://github.com/KILTprotocol/ticket/issues/2627

### CODE REVIEW

File: src/utilities/identities/identities.ts [hunk 1]
```diff
@@ -58,7 +58,7 @@ export function useIdentities(): IdentitiesContextType {
   return useContext(IdentitiesContext);
 }
 
-async function getCurrentIdentity(): Promise<string | null> {
+export async function getCurrentIdentity(): Promise<string | null> {
```
Comment (lukeg90) [inline src/utilities/identities/identities.ts:61]: Is it imported anywhere?

File: src/channels/CredentialChannels/backgroundCredentialChannel.ts [hunk 2]
```diff
@@ -37,6 +38,16 @@ export async function showCredentialPopup(
...
       }
+
+      // the DID to use for signing could be predetermined by the dApp, if we have a matching identity we’ll use it
+      const { owner } = content.claim;
+      const identities = await getIdentities();
+      for (const { address, did } of Object.values(identities)) {
+        if (did === owner) {
+          await setCurrentIdentity(address);
+        }
+      }
```
Comment (lukeg90) [inline src/channels/CredentialChannels/backgroundCredentialChannel.ts:49]: Does the loop need to continue after the matching identity is found?

**Review by lukeg90 — COMMENTED** [review]
Does it make sense to add a new test for `SignQuote` for when the owner is not included in the Terms?

File: src/utilities/identities/identities.ts (hunk 1 above)
Comment (arty-name) [reply src/utilities/identities/identities.ts:61]: Good catch! Not anymore, I only needed that for an interim solution.

File: src/channels/CredentialChannels/backgroundCredentialChannel.ts (hunk 2 above)
Comment (arty-name) [reply src/channels/CredentialChannels/backgroundCredentialChannel.ts:49]: No, but it doesn’t really hurt, while completely avoiding the concerns about the flow control

**Comment by arty-name** [comment]
> Does it make sense to add a new test for SignQuote for when the owner is not included in the Terms?

Turns out the test we had for it was kind of broken

**Review by lukeg90 — APPROVED** [review]

**Pull request merged**